.evergreen/evergreen.yml.in

```diff
@@ -185,21 +196,61 @@ functions:
           source .evergreen/setup-env.sh
           npm --unsafe-perm=true run bootstrap-ci -- --ignore-prepublish
           npm run test-ci
-          echo "Creating coverage tarball..."
-          tar cvzf coverage.tgz coverage
+          echo "Archiving current coverage result..."
+          tar cvzf nyc-output.tgz .nyc_output
           }
         env:
           AWS_AUTH_IAM_ACCESS_KEY_ID: ${devtools_ci_aws_key}
           AWS_AUTH_IAM_SECRET_ACCESS_KEY: ${devtools_ci_aws_secret}
+    - command: s3.put
+      params:
+        aws_key: ${aws_key}
+        aws_secret: ${aws_secret}
+        local_file: src/nyc-output.tgz
+        remote_file: mongosh/binaries/${revision}/${revision_order_id}/nyc-output-${build_variant}-${mongosh_test_id}.tgz
+        bucket: mciuploads
+        permissions: public-read
+        content_type: application/x-gzip
+
+  check_coverage:
+    <% for (let buildVariant of ALL_UNIT_TEST_BUILD_VARIANTS) {
+        for (let unitTest of ALL_UNIT_TESTS) { %>
+    - command: s3.get
+      params:
+        aws_key: ${aws_key}
+        aws_secret: ${aws_secret}
+        local_file: src/nyc-output-<% out(buildVariant) %>-<% out(unitTest.id) %>.tgz
+        remote_file: mongosh/binaries/${revision}/${revision_order_id}/nyc-output-<% out(buildVariant) %>-<% out(unitTest.id) %>.tgz
+        bucket: mciuploads
+    - command: shell.exec
+      params:
+        working_dir: src
+        shell: bash
+        script: tar xvzf nyc-output-<% out(buildVariant) %>-<% out(unitTest.id) %>.tgz
```

Comment:
Can this be a script with `set -e`, like the other bash scripts? We would want to know about if this fails, I think